packages/nuxt/src/core/builder.ts

```diff
@@ -19,6 +19,11 @@ export async function build (nuxt: Nuxt) {
   await generateApp()
 
   if (nuxt.options.dev) {
+    const noAppFolder = nuxt.options.srcDir === nuxt.options.rootDir
+    if (noAppFolder && !nuxt.options.test) {
+      logger.warn('You are using the Nuxt v3 directory structure. We recommend moving your application code inside an /app directory in Nuxt v4. https://nuxt.com/blog/v4#%EF%B8%8F-new-project-structure')
```

Comment:
What about something like:

```suggestion
      logger.info('Using legacy directory structure. https://nuxt.com/blog/v4#%EF%B8%8F-new-project-structure')
```

... and we can use a short url instead of the blog link.

wdyt @atinux?

Reply:
Agree, see my last code suggestion